app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -53,58 +53,102 @@ class TransferSendManager @Inject constructor(
+            uploadWorkerScheduler.scheduleWork(uuid)
+            _sendActionResult.update {
+                val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
+                SendActionResult.Success(totalSize)
+            }
+        }.onFailure { exception ->
+            SentryLog.e(TAG, "Failed to start the upload", exception)
+            _sendActionResult.update { SendActionResult.Failure }
         }
     }
 
-    private fun CoroutineScope.requestAppIntegrityToken(
-        appIntegrityManager: AppIntegrityManager,
-        newUploadSession: NewUploadSession,
+    //region App Integrity
+    private suspend inline fun withIntegrityToken(
+        onSuccess: (attestationToken: String) -> Unit,
+        onRefused: () -> Unit = {},
+        onFailure: (exception: Throwable) -> Unit = {},
     ) {
-        appIntegrityManager.requestClassicIntegrityVerdictToken(
-            onSuccess = { token ->
-                SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful $token")
-                getApiIntegrityVerdict(appIntegrityManager, token, newUploadSession)
-            },
-            onFailure = ::setFailedIntegrityResult,
-        )
+        runCatching {
+            var attestationToken: String? = null
+
+            coroutineScope {
+                appIntegrityManager.getChallenge(
+                    onSuccess = { launch { attestationToken = requestAppIntegrityToken(appIntegrityManager) } },
+                    onFailure = {},
+                )
+            }
+
+            attestationToken?.let(onSuccess) ?: onRefused()
+        }.onFailure {
+            onFailure.invoke(it)
+        }
     }
 
-    private fun CoroutineScope.getApiIntegrityVerdict(
-        appIntegrityManager: AppIntegrityManager,
-        appIntegrityToken: String,
-        newUploadSession: NewUploadSession,
-    ) {
-        launch {
-            appIntegrityManager.getApiIntegrityVerdict(
-                integrityToken = appIntegrityToken,
-                packageName = BuildConfig.APPLICATION_ID,
-                targetUrl = sharedApiUrlCreator.createUploadContainerUrl,
-                onSuccess = { attestationToken ->
-                    SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "Api verdict check")
-                    Log.i(APP_INTEGRITY_MANAGER_TAG, "getApiIntegrityVerdict: $attestationToken")
-                    _integrityCheckResult.value = AppIntegrityResult.Success
-                    sendTransfer(attestationToken, newUploadSession)
+    private suspend fun requestAppIntegrityToken(appIntegrityManager: AppIntegrityManager): String? {
+        var attestationToken: String? = null
+
+        coroutineScope {
+            appIntegrityManager.requestClassicIntegrityVerdictToken(
+                onSuccess = { token ->
+                    SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful $token")
```

Comment:
I'd rather not put the token on sentry

Reply:
I agree, I removed it from the sentry altogether